_posts/RhymesWithMecca/2016-02-03-BeccasChapter5.md

```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+author: RhymesWithMecca
+title: "Becca's Chapter 5 Exercises"
+---
+
+Here is Exercise 1:
+<iframe src="https://trinket.io/embed/python/79b285a420" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Here is Exercise 2:
+<iframe src="https://trinket.io/embed/python/22b00b018c" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Exercise 1 Thoughts:
+This one took me a while.  First I had to figure out the proper order of the ifs, try/except, and while loop.  I needed to put the if and break statement before turning the input into a float.  I screwed up my indentation in the try conditional.  At one point, it took inputs but always printed the error message in the except loop.
+
+Then I got stuck and went to sleep.  The next day, I tried it again with functions, but it was printing something like <function sum> and I couldn't figure out why.  The weird input messages were still occurring.  I attempted to use global x for the first time, but no dice.
+
+I then looked ahead on the schedule and saw that Chapter 8 is about lists.  This looked promising.  Midway through the chapter, I saw a part about remaking the original summing program and thought that might be useful.  I used that to help me write my program.
```

Comment:
What the lists allow you to do is use the min() and max() functions to get those inputs.  But if you think about what they're doing, they iterate through the list and remember the max and min.  Since you're already iterating through the list, so to speak, each time the user inputs a number, you can remember the max and min too- that's how to solve Exercise 2 without lists.